### DISCUSSION TITLE
Get object response without AutoInvokeKernelFunctions.

### DISCUSSION THREAD

**Discussion by SalomonHenke** [microsoft/semantic-kernel#8308] [Q&A] [answered]
Hey all. Not sure if I have missed something, but is there an obvious way for me to do something like this: 
```
var prompt = "Generate the fields for my object please :)"
MyObjectThatIWantTheModelToPopulate myobj = kernel.HokusPokus(prompt);
```

I am able to do this via. the plugins e.g. `[KernelFuncton]` with a complex object as the parameter with description on the items. Im just wondering if there is a way for me to do this without having to rely on chat completion and the AutoInvokeKernelFunctions as there are cases where I know exactly what I need it to generate.

I can call a plugin directly, but in that case it wants me to provide function param, that the model generates itself on chatCompletion tool calling.

**Comment by moonbox3** [comment]
Hi @SalomonHenke, if I am reading this correctly, you do want to invoke the function and not have the LLM put a final answer once function invocation is complete. Is that true?

**Reply by SalomonHenke** [reply]
Hey @moonbox3, I guess that's one way to put it. I'm really trying to mimic how I used to do it with the older models, like InvokeSpecificModel, where I could define a return model, and that was it—no chat completion, just an object that gets populated based on the request.

For example, I'd give the LLM a prompt like, "Create a poem about dogs," and it would return just the structure and nothing else, like this:

```
Poem
{
     Title,
     AnotherParameter,
     AThirdParameter
}
```
So essentially, the syntax would go something like this:

Poem myPoem = llm.InvokeCreatePoemFunction(prompt);

I guess I can somewhat achieve this with planners, but I haven't had the best luck with consistency using those. Really just wondering if there is a syntax that does this for a singular function.

**Comment by SalomonHenke** [comment]
Not answered. I might not be explaining it well enough. 

Essentially I want to achieve function calling like it was possible via openAI API previously. e.g. 

Define a structure I want returned e.g. 

```
public class Poem
{
    public string Title { get; set; }
    public string PoemText { get; set; }
    public int AnotherField { get; set; }
    public int AndSoOn { get; set; }
}

and say 
var prompt = "a poem about my struggle to explain what I want to achieve in my github Q&A question".
var generatedPoem = myAIFunctionCall(prompt);
```

And I am able to do this via. ChatCompletion and AutoInvoke, but it seems wasteful as I know I want to generate a Poem and i am not interested in any other tool call. I essentially kill the flow the second my plugin is called by invoking a delegate to return the generated request object as that is the only thing im interested in. This doesn't seem like the way to do  it. Or perhaps my use-case isn't supported in a convenient way?

**Reply by markwallace-microsoft** [reply, marked as answer]
@SalomonHenke if I understand correctly you want the model to generate the function call args (i.e. request object) and that is the outcome you want for your function. If that's correct you can use the `IAutoFunctionInvocationFilter` to achieve what you want i.e. when the filter gets invoked you have access to the arguments the model generated and you can set then as the result and terminate the function calling loop.

**Reply by SalomonHenke** [reply]
Alright this makes sense, but this also means there isnt a build in way to do the direct "class structure builder" thingy?  Thanks for the answer @markwallace-microsoft

**Reply by markwallace-microsoft** [reply]
Here is a link to a sample: https://github.com/microsoft/semantic-kernel/blob/53dcf39eb65b55723550a93b22eb837ab96592c4/dotnet/samples/Concepts/Filtering/AutoFunctionInvocationFiltering.cs#L15
